### PR TITLE
Copy and delete cubes in Molecule correctly

### PR DESCRIPTION
**Pull request by psavery**
This potentially fixes a memory leak in the Molecule
class for when the destructor is called and the cubes
are never deleted.

It also fixes any issues with copying the cubes via
the copy constructor or assignment operator.

### CODE REVIEW

**Comment by psavery** [comment]
Also, I noticed that the copy constructor and assignment operator do not copy over all of the data... I wasn't sure if this was intentional or not.

Here are the members that I noticed are not copied:
m_graph
m_data
m_coordinates3d
m_vibrationFrequencies
m_vibrationIntensities
m_vibrationLx

File: avogadro/core/molecule.cpp [hunk 1]
```diff
@@ -52,6 +52,12 @@ Molecule::Molecule(const Molecule &other)
     Mesh *m = addMesh();
     *m = *other.mesh(i);
   }
+
+  // Copy over any cubes
+  for(Index i = 0; i < other.cubeCount(); ++i) {
```
Comment (cryos) [inline avogadro/core/molecule.cpp]: Should be a space after the for keyword

File: avogadro/core/molecule.cpp [hunk 2]
```diff
@@ -79,6 +85,14 @@ Molecule& Molecule::operator=(const Molecule& other)
       Mesh *m = addMesh();
       *m = *other.mesh(i);
     }
+
+    clearCubes();
+
+    // Copy over any cubes
+    for(Index i = 0; i < other.cubeCount(); ++i) {
```
Comment (cryos) [inline avogadro/core/molecule.cpp]: As above

**Comment by cryos** [comment]
We should look at the others, I don't think it was intentional.

**Comment by psavery** [comment]
Okay. I will submit that very soon.

Are you okay with the extra commit that was a part of this one? I was trying to get rid of it... it seems to be unnecessary

**Comment by cryos** [comment]
It would have been good to remove the extra commit, but it is merged now and not critical.

**Pull request merged**